core/src/main/java/org/kiwix/kiwixmobile/core/page/bookmark/BookmarksFragment.kt

```diff
@@ -1,15 +1,20 @@
 package org.kiwix.kiwixmobile.core.page.bookmark
 
 import org.kiwix.kiwixmobile.core.R
-import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.base.BaseActivity
+import org.kiwix.kiwixmobile.core.extensions.ActivityExtensions.coreActivityComponent
 import org.kiwix.kiwixmobile.core.extensions.ActivityExtensions.viewModel
-import org.kiwix.kiwixmobile.core.page.PageActivity
+import org.kiwix.kiwixmobile.core.page.PageFragment
 import org.kiwix.kiwixmobile.core.page.adapter.PageAdapter
 import org.kiwix.kiwixmobile.core.page.adapter.PageDelegate.PageItemDelegate
 import org.kiwix.kiwixmobile.core.page.bookmark.viewmodel.BookmarkViewModel
 
-class BookmarksActivity : PageActivity() {
-  override val pageViewModel by lazy { viewModel<BookmarkViewModel>(viewModelFactory) }
+class BookmarksFragment : PageFragment() {
+  override val pageViewModel by lazy {
+    requireActivity().viewModel<BookmarkViewModel>(
```

Comment:
these viewmodels are scoped to the activity so will stick around much longer than we want, we probably want these scoped only to the fragments